schema_editor/test/spec/schemas/schemas-service.spec.js

```diff
@@ -32,12 +32,35 @@ describe('ase.schemas:Schemas', function () {
         }];
         var dataFormSchemaDef = Schemas.definitionFromSchemaFormData(schemaFormData);
         expect(dataFormSchemaDef).toEqual(jasmine.objectContaining({
-            required: jasmine.anything(),
             properties: jasmine.anything(),
             type: 'object'
         }));
     });
 
+    it('should serialize a textOptions key in text fields', function () {
+        var schemaFormData = [{
+            fieldType: 'text',
+            isRequired: false,
+            fieldTitle: 'Text',
+            textOptions: 'datetime'
+        }];
+        var dataFormSchemaDef = Schemas.definitionFromSchemaFormData(schemaFormData);
+        expect(dataFormSchemaDef.properties.Text.textOptions).toEqual(schemaFormData[0].textOptions);
+    });
+
+    it('should serialize an isSearchable key', function () {
+        var schemaFormData = [{
+            fieldType: 'text',
+            isRequired: false,
+            isSearchable: true,
+            fieldTitle: 'Text',
+            textOptions: 'datetime'
+        }];
+        var dataFormSchemaDef = Schemas.definitionFromSchemaFormData(schemaFormData);
+        expect(dataFormSchemaDef.properties.Text.isSearchable)
+            .toEqual(schemaFormData[0].isSearchable);
+    });
+
```

Comment:
One more test would be good, to test that `required` is not included if `isRequired === false` for all fields.